build-system/tasks/extension-generator/index.js

```diff
@@ -0,0 +1,291 @@
+  </tr>
+  <tr>
+    <td class="col-fourty"><strong><a href="https://www.ampproject.org/docs/guides/responsive/control_layout.html">Supported Layouts</a></strong></td>
+    <td>FILL THIS IN</td>
+  </tr>
+  <tr>
+    <td width="40%"><strong>Examples</strong></td>
+    <td>FILL THIS IN</td>
+  </tr>
+</table>
+
+## Behavior
+
+FILL THIS IN. What does this extension do?
+
+## Attributes
+
+FILL THIS IN. Does this extension allow for properties to configure?
+
+## Validation
+See [${name} rules](https://github.com/ampproject/amphtml/blob/master/extensions/${name}/0.1/validator-${name}.protoascii) in the AMP validator specification.
+`;
+}
+
+function getJsTestExtensionFile(name) {
+  const className = pascalCase(name);
+  return `/**
+ * Copyright ${year} The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {${className}} from '../${name}';
+
+describes.realWin('${name}', {
+  amp: {
+    extensions: ['${name}'],
+  }
+}, env => {
+
+  let win;
+  let element;
+
+  beforeEach(() => {
+    win = env.win;
+    element = win.document.createElement('${name}');
+    win.document.body.appendChild(element);
+  });
+
+  it('should have hello world when built', () => {
+    expect(element.textContent).to.equal('hello world');
+  });
+});
+`;
+}
+
+function getJsExtensionFile(name) {
+  const className = pascalCase(name);
+  return `/**
+ * Copyright ${year} The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Layout} from '../../../src/layout';
+
+export class ${className} extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {string} */
+    this.myText_ = 'hello world';
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return layout == Layout.CONTAINER || layout == Layout.RESPONSIVE;
+  }
+
+  /** @override */
+  buildCallback() {
```

Comment:
what's a good example there? also what's the best layout for just text component?